File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/illustration/illustration-lookup-model.ts
import { IProductModelIllustrationLookupModel } from 'app/models/data/entities/production/product-model-illustration/product-model-illustration-lookup-model';

export interface IIllustrationLookupModel {
  illustrationID: number;
  diagram: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class IllustrationLookupModel implements IIllustrationLookupModel {
  illustrationID: number;
  diagram: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<IllustrationLookupModel>) {
    if (!!init) {
      Object.assign<IllustrationLookupModel, Partial<IllustrationLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Production/Illustration/Commands/Queries/GetIllustrations/IllustrationLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.Production.Illustration.Queries.GetIllustrations
{
    public partial class IllustrationLookupModel : IHaveCustomMapping
    {
        public int IllustrationID { get; set; }
        public string Diagram { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.Illustration, IllustrationLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<IllustrationLookupModel, BaseIllustration>().IgnoreMissingDestinationMembers();
        }
    }
}
